dot/core/service.go

```diff
@@ -462,6 +473,25 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 	return nil
 }
 
+func (s *Service) notifyRuntimeUpdated(version runtime.Version) {
+	s.runtimeChangedLock.RLock()
+	defer s.runtimeChangedLock.RUnlock()
+
+	if len(s.runtimeChanged) == 0 {
+		return
+	}
+
+	logger.Info("notifying runtime updated chans...", "chans", s.runtimeChanged)
+	for _, ch := range s.runtimeChanged {
+		go func(ch chan<- runtime.Version) {
+			select {
+			case ch <- version:
```

Comment:
why is the channel being written to as a select case?

Reply:
Good question, no reason, so I've removed the select case.